src/WebJobs.Script.WebHost/Middleware/AllowSynchronousIOMiddleware.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class AllowSynchronousIOMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public AllowSynchronousIOMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            context.Features.Get<IHttpBodyControlFeature>().AllowSynchronousIO = true;
```

Comment:
Are we guaranteed to have this feature in the collection in all contexts? Should we validate we got it back?

Reply:
I think so. But I can pull and check first.

Reply:
Just added a `?` here.

Reply:
fixed